### PR TITLE
feat: add proof submitter address

### PR DESCRIPTION
**Pull request by wadealexc**
From Slack:

I have a small feature request for PEPE -- allowing a pod owner to approve a second address to start checkpoints and verify validator credentials. This change would be a new state variable plus a single, small method in EigenPod.sol that allows the pod owner to change the `proofSubmitter` address.

I think this would be super helpful for things like our CLI, which allows you to pass in a private key to submit transactions. However, Stakers will not want to submit their pod owner's private key to the CLI, because that key has access to literally all of their funds. We need a way for them to set up a hot wallet to accomplish their daily proving tasks.

### CODE REVIEW

**Review by gpsanant — APPROVED** [review]

File: src/contracts/pods/EigenPodStorage.sol [hunk 1]
```diff
@@ -71,10 +71,16 @@ abstract contract EigenPodStorage is IEigenPod {
     /// @notice The current checkpoint, if there is one active
     Checkpoint internal _currentCheckpoint;
 
+    /// @notice An address with permissions to call `startCheckpoint` and `verifyCheckpointProofs`, set
+    /// by the podOwner. This role exists to allow a podOwner to designate a hot wallet that can call
+    /// these methods, allowing the podOwner to remain a cold wallet that is only used to manage funds.
+    /// @dev If this address is NOT set, only the podOwner can call `startCheckpoint` and `verifyCheckpointProofs`
```
Comment (8sunyuan) [inline src/contracts/pods/EigenPodStorage.sol:77]: typo here, should be `verifyWithdrawalCredentials`

File: src/contracts/interfaces/IEigenPod.sol [hunk 2]
```diff
@@ -186,10 +189,25 @@ interface IEigenPod {
...
+
     /*******************************************************************************
                                    VIEW METHODS
     *******************************************************************************/
 
+    /// @notice An address with permissions to call `startCheckpoint` and `verifyCheckpointProofs`, set
+    /// by the podOwner. This role exists to allow a podOwner to designate a hot wallet that can call
+    /// these methods, allowing the podOwner to remain a cold wallet that is only used to manage funds.
+    /// @dev If this address is NOT set, only the podOwner can call `startCheckpoint` and `verifyCheckpointProofs`
+    function proofSubmitter() external view returns (address);
```
Comment (8sunyuan) [inline src/contracts/interfaces/IEigenPod.sol:209]: same thing here too with comments

Comment (wadealexc) [reply src/contracts/interfaces/IEigenPod.sol:209]: [b23f172](https://github.com/Layr-Labs/eigenlayer-contracts/pull/629/commits/b23f1720212c84c44874c5247baa59bdf3152770)

File: src/contracts/pods/EigenPodStorage.sol (hunk 1 above)
Comment (wadealexc) [reply src/contracts/pods/EigenPodStorage.sol]: [b23f172](https://github.com/Layr-Labs/eigenlayer-contracts/pull/629/commits/b23f1720212c84c44874c5247baa59bdf3152770)

**Pull request merged**